src/Avalonia.Base/Visual.cs

```diff
@@ -667,8 +667,14 @@ private void SetVisualParent(Visual? value)
 
             if (_visualParent is IRenderRoot || _visualParent?.IsAttachedToVisualTree == true)
             {
-                var root = this.FindAncestorOfType<IRenderRoot>() ??
-                    throw new AvaloniaInternalException("Visual is atached to visual tree but root could not be found.");
+                var root = this.FindAncestorOfType<IRenderRoot>();
```

Comment:
It does not really solve the problem, maybe it's better like this:
 ```
var root = _visualParent.VisualRoot ??
        throw new AvaloniaInternalException("Visual is atached to visual tree but root could not be found.");
```

Reply:
Yeah I think just ignoring this error here is going to cause other, more serious and hard to diagnose problems down the line. Fixing an issue like this really needs to start with a unit test to understand the cause of the problem rather than papering over it.

Reply:
I think if we check for the visual root in TransitioningContentControl.UpdateContentWithTransition method, it might solve the problem. I.e. after async call and before assigning the content.  @hez2010